runtime/src/parachains.rs

```diff
@@ -381,7 +381,7 @@ impl<T: Trait> Module<T> {
 
 			// track which voters have voted already, 1 bit per authority.
 			let mut track_voters = bitvec![0; authorities.len()];
-			for (auth_id, validity_attestation) in &candidate.validity_votes {
+			for (auth_index, validity_attestation) in &candidate.validity_votes {
```

Comment:
rather than pass in `auth_index` we would ideally pass in a serialized form of the bit vector used here. But passing in the index instead of a key is a good first step.

Reply:
I haven't exactly grasped the idea. Currently we have `validity_votes` in the form of `Vec<(ValidatorIndex, ValidityAttestation)>`. Do you propose to split it into `BitVec` plus `Vec<ValidityAttestation>`?

Reply:
Yes, that's right

Reply:
https://github.com/paritytech/polkadot/issues/254